Add FAQ entry about supply chain attacks on libraries (#2146)
chainguard-dev/edu: content/chainguard/libraries/faq.md

toc: true
---

## What security issues can Chainguard Libraries prevent?

As detailed in the [background](/chainguard/libraries/overview/#background) and
[introduction](/chainguard/libraries/overview/#introduction) Chainguard
Libraries are built directly from source in the Chainguard Factory and the
resulting binaries are directly provided to you by Chainguard. Chainguard
operates the whole supply chain for the package lifecycle as one reliable,
secure partner. You can therefore avoid issues from the following software

> supply chain attacks surface points:

Author: Fixed

* Build pipeline
* Build system
* Dependency injection
* Bypass of CI/CD systems
* Library distribution
* Library consumption

> More information about the stages is available on the [Supply chain Levels for

Author: Fixed

Software Artifacts (SLSA) website](https://slsa.dev/).

Following are examples that exploited issues in the relevant supply
chain:

> **Example: Ultralytics Python project**

Reviewer: I don't think we need to include "Example", it's assumed

```suggestion
**Ultralytics Python project**
```

Author: Done here and for others

> **Example: Ultralytics Python project**

Author: H3 works well... 

Yes.. this in the Chainguard Libraries FAQ ... and the problem is more prevalent outside the Java ecosystem.. also that is how PMM and others have messaged so far so we should continue..

* Attackers compromised Ultralytics' GitHub Actions workflow, injecting malware
  into PyPI package releases.
* Malicious versions proliferated (8.3.41, 8.3.42, 8.3.45, 8.3.46).

> * Affected a widely-used AI library with ~60 million downloads included a cryptominer.

Reviewer: ```suggestion
* Affected a widely-used AI library with ~60 million downloads that included a cryptominer.
```

Author: fixed

> * No source code was included and Chainguard Libraries would therefore block.

Reviewer: I think this needs to be spelled out more for people less familiar with the product. I am not sure if I phrased it very elegantly here, so feel free to iterate

```suggestion
* No source code was included: the use of Chainguard Libraries, based on source code, would have prevented the attack.
```

Author: Fixed

* See also [PyPI attack analysis](https://blog.pypi.org/posts/2024-12-11-ultralytics-attack-analysis/) and

>   [bleepingcomputer blog post](https://www.bleepingcomputer.com/news/security/ultralytics-ai-model-hijacked-to-infect-thousands-with-cryptominer/)

Reviewer: Consistency with bullet point punctuation

```suggestion
  [bleepingcomputer blog post](https://www.bleepingcomputer.com/news/security/ultralytics-ai-model-hijacked-to-infect-thousands-with-cryptominer/).
```

Author: ✅

**Example: Lottie Player**

> * Hackers gained access to NPM registry by compromising a developer authentication token.

Reviewer: I would say the same about NPM here

Author: not sure what you mean

> * Used token to upload a compromised version of Lottie Player.

Reviewer: ```suggestion
* Token used to upload a compromised version of Lottie Player.
```

Author: Fixed

* The malicious package would drain crypto wallet funds.

> * No source code was included and Chainguard Libraries would therefore block.

Reviewer: ```suggestion
* No source code was included: the use of Chainguard Libraries, based on source code, would have prevented the attack.
```

Author: ✅

> * See also [npm package Lottie-Player compromised in supply chain attack, Nov 2024](https://www.infosecurity-magazine.com/news/npm-package-lottieplayer-supply/)  

Reviewer: ```suggestion
* See also [npm package Lottie-Player compromised in supply chain attack, Nov 2024](https://www.infosecurity-magazine.com/news/npm-package-lottieplayer-supply/).
```

Author: ✅

**Example: MavenGate**

* MavenGate exploits abandoned Java library domains.
* Vulnerabilities in Maven dependency management allowed unauthorized package replacements.

> * All Maven-based software, including Gradle and Ant, is compromised.

Reviewer: ```suggestion
* All Maven-based software, including Gradle and Ant, was compromised.
```

* See also [thehackernews article](https://thehackernews.com/2024/01/hackers-hijack-popular-java-and-android.html) and 

>   [oversecured blog post](https://blog.oversecured.com/Introducing-MavenGate-a-supply-chain-attack-method-for-Java-and-Android-applications/)

Reviewer: ```suggestion
  [oversecured blog post](https://blog.oversecured.com/Introducing-MavenGate-a-supply-chain-attack-method-for-Java-and-Android-applications/).
```

Author: ✅
